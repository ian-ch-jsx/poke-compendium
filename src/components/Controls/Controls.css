.controls {
  margin: auto 0;
  display: flex;
  justify-content: center;
  align-items: flex-end;
}

.select {
  background-color: #8ec5fc;
  background-image: linear-gradient(62deg, #8ec5fc 0%, #e0c3fc 100%);
  border-radius: 50px;
  border: 0px;
  filter: drop-shadow(5px 5px 4px #000000);
  height: 30px;
  border: 0;
}

#type-label,
#order-label {
  color: white;
}
