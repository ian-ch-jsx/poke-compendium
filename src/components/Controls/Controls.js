import './Controls.css';
import { Select, MenuItem, InputLabel, FormControl } from '@mui/material';

export default function Controls({
  query,
  setQuery,
  setLoading,
  order,
  setOrder,
  selectedType,
  setSelectedType,
}) {
  return (
    <div className="controls">
      <FormControl variant="standard" sx={{ m: 1, minWidth: 120 }}>
        <InputLabel id="type-label" htmlFor="uncontrolled-native">
          Type
        </InputLabel>
        <Select
          className="select"
          labelId="type-label"
          label="type"
          value={selectedType}
          onChange={(e) => setSelectedType(e.target.value)}
        >
          <MenuItem value="all">All</MenuItem>
          <MenuItem value="bug">Bug</MenuItem>
          <MenuItem value="dark">Dark</MenuItem>
          <MenuItem value="dragon">Dragon</MenuItem>
          <MenuItem value="electric">Electric</MenuItem>
          <MenuItem value="fairy">Fairy</MenuItem>
          <MenuItem value="fighting">Fighting</MenuItem>
          <MenuItem value="fire">Fire</MenuItem>
          <MenuItem value="flying">Flying</MenuItem>
          <MenuItem value="ghost">Ghost</MenuItem>
          <MenuItem value="grass">Grass</MenuItem>
          <MenuItem value="ground">Ground</MenuItem>
          <MenuItem value="ice">Ice</MenuItem>
          <MenuItem value="normal">Normal</MenuItem>
          <MenuItem value="poison">Poison</MenuItem>
          <MenuItem value="psychic">Psychic</MenuItem>
          <MenuItem value="rock">Rock</MenuItem>
          <MenuItem value="steel">Steel</MenuItem>
          <MenuItem value="water">Water</MenuItem>
        </Select>
      </FormControl>
      <FormControl variant="standard" sx={{ m: 1, minWidth: 120, color: '.menu' }}>
        <InputLabel id="order-label" variant="standard" htmlFor="uncontrolled-native">
          order
        </InputLabel>
        <Select
          className="select"
          labelId="order-label"
          value={order}
          onChange={(e) => setOrder(e.target.value)}
        >
          <MenuItem value="asc">Ascending</MenuItem>
          <MenuItem value="desc"> Descending</MenuItem>
        </Select>
      </FormControl>
      <input
        type="text"
        placeholder="search by name"
        value={query}
        onChange={(e) => {
          setQuery(e.target.value);
        }}
      />
      <button onClick={() => setLoading(true)}>Search</button>
    </div>
  );
}
