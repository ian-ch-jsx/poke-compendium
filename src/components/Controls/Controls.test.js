import { render } from '@testing-library/react';
import Controls from './Controls';
import { Select } from '@mui/material';

test('starts on default ascending order', () => {
  render(<Controls />);
  const controls = render(<Select className="select" labelId="order-label" value="asc" />);
  expect(controls).toMatchSnapshot();
});
