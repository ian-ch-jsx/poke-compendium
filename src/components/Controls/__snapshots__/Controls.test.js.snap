// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`starts on default ascending order 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="controls"
      >
        <div
          class="MuiFormControl-root css-1869usk-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="uncontrolled-native"
            id="type-label"
          >
            Type
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl select css-1480iag-MuiInputBase-root-MuiInput-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="type-label"
              class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiFormControl-root css-9n0v02-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-aqpgxn-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="uncontrolled-native"
            id="order-label"
          >
            order
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl select css-1480iag-MuiInputBase-root-MuiInput-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="order-label"
              class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
        <input
          placeholder="search by name"
          type="text"
          value=""
        />
        <button>
          Search
        </button>
      </div>
    </div>
    <div>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary select css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="order-label"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="asc"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-nnbavb"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary select css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="order-label"
        class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          ​
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="asc"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-nnbavb"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
